src/main/java/uk/ac/ebi/eva/commons/core/models/Variant.java

```diff
@@ -0,0 +1,240 @@
+     * deleted nucleotide is in position 6, the start is position 6</li>
+     * </ul>
+     */
+    private final int start;
+
+    /**
+     * Position where the genomic variation ends.
+     * <ul>
+     * <li>SNVs have the same start and end positions</li>
+     * <li>Insertions end in the first present position: if the last nucleotide
+     * is inserted in position 9, the end is position 10</li>
+     * <li>Deletions ends in the last previously present position: if the last
+     * deleted nucleotide is in position 9, the end is position 9</li>
+     * </ul>
+     */
+    private final int end;
+
+    /**
+     * Reference allele.
+     */
+    private final String reference;
+
+    /**
+     * Alternate allele.
+     */
+    private final String alternate;
+
+    /**
+     * Set of identifiers used for this genomic variation.
+     */
+    private final Set<String> ids;
+
+    /**
+     * Unique identifier following the HGVS nomenclature.
+     */
+    private final Map<String, Set<String>> hgvs;
+
+    /**
+     * Information specific to each file the variant was read from, such as samples or statistics.
+     */
+    private final Map<String, IVariantSourceEntry> sourceEntries;
+
+    public Variant(String chromosome, int start, int end, String reference, String alternate) {
+        if (start > end && !(reference.equals("-"))) {
+            throw new IllegalArgumentException("End position must be greater than the start position");
+        }
+
+        this.chromosome = (chromosome != null) ? chromosome : "";
+        this.start = start;
+        this.end = end;
+        this.reference = (reference != null) ? reference : "";
+        this.alternate = (alternate != null) ? alternate : "";
+
+        this.ids = new HashSet<>();
+        this.hgvs = new HashMap<>();
+        if (getType() == VariantType.SNV) { // Generate HGVS code only for SNVs
+            Set<String> hgvsCodes = new HashSet<>();
+            hgvsCodes.add(chromosome + ":g." + start + reference + ">" + alternate);
+            this.hgvs.put("genomic", hgvsCodes);
+        }
+        sourceEntries = new HashMap<>();
+    }
+
+    public int getLength() {
+        return Math.max(this.reference.length(), this.alternate.length());
+    }
+
+    public VariantType getType() {
+        if (reference.length() == alternate.length()) {
+            if (getLength() > 1) {
+                return VariantType.MNV;
+            } else {
+                return VariantType.SNV;
+            }
+        } else if (getLength() <= SV_THRESHOLD) {
+            /*
+            * 3 possibilities for being an INDEL:
+            * - The value of the ALT field is <DEL> or <INS>
+            * - The REF allele is not . but the ALT is
+            * - The REF allele is . but the ALT is not
+            * - The REF field length is different than the ALT field length
+            */
+            return VariantType.INDEL;
+        } else {
+            return VariantType.SV;
```

Comment:
This should be discussed outside of the pr as it is REALLY relevant and not exclusive of this pr.